### PR TITLE
feat(corel): Add review changes screen 

### PR DESCRIPTION
**Pull request by pedrobonamin**
### Description
Adds review details screen to content releases.
This screen allows users to compare the version document with the published one, reusing the diff components that are currently part of the studio. 

It updates some of the Diff components with selectors to override the default style.

Adds a new hook in the preview store, that allows us to listen to entire documents without adding new listeners by reusing the mechanisms built into the preview store. [useObserveDocument](https://github.com/sanity-io/sanity/pull/7155/files#diff-41f33a8452d3d9209a5e7cd3aed995639d4bc1279a984d34ad647b105722dd1e)
The publish and version documents are live updated, any change happening to any of the documents will be immediately reflected in the UI.

**Improvements**

Add virtualisation to the documents review.
Add search to review screen.
Update UI for documents with no changes.
Update UI for new documents.

### CODE REVIEW

File: packages/sanity/src/core/releases/tool/detail/ReleaseDetail.tsx [hunk 1]
```diff
@@ -37,6 +52,20 @@ export const ReleaseDetail = () => {
...
+    })
+  }, [router])
+
+  const navigateToOverview = useCallback(() => {
+    router.navigate({
+      ...router.state,
+      _searchParams: [],
+    })
+  }, [router])
+
```
Comment (RitaDias) [inline packages/sanity/src/core/releases/tool/detail/ReleaseDetail.tsx:68]: yessss, yeEEEEESSSS

File: packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      wrap(buildDocumentForDiffInput(baseDocument), null),
+      wrap(buildDocumentForDiffInput(document), null),
+    ) as ObjectDiff
+
+    if (!diff.isChanged) return {changesList: [], rootDiff: null}
+    const changeList = buildChangeList(schemaType, diff, [], [], {})
+    return {changesList: changeList, rootDiff: diff}
+  }, [baseDocument, document, schemaType])
+
+  const isChanged = rootDiff?.isChanged || false
```
Comment (RitaDias) [inline packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx:45]: really small nit: I'm having a hard time mapping this /but/ I think that might be best to to use here is the `??` operator? 🤔

**Review by RitaDias — APPROVED** [review]
Small comment but nothing blocking 🚀

File: packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx (hunk 2 above)
Comment (pedrobonamin) [reply packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx:45]: Wdyt of 
```suggestion
  const isChanged = !!rooitDiff?.isChanged
```

Comment (RitaDias) [reply packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx:45]: Love it :)

File: packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx
Comment (jordanl17) [inline packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx:1]: @pedrobonamin thoughts on adding some tests for a few different diffs on this screen to do a brief, overall integration test of the associated components?

File: packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {type SanityDocument} from '@sanity/types'
+import {Stack} from '@sanity/ui'
+// eslint-disable-next-line camelcase
```
Comment (jordanl17) [inline packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx:3]: Is this disable required?

File: packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx
Comment (pedrobonamin) [reply packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx:1]: Yeah, missed that, will add tests for this, thanks!

File: packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx (hunk 3 above)
Comment (pedrobonamin) [reply packages/sanity/src/core/releases/tool/detail/ReleaseReview.tsx:3]: Nope, will remove

File: packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  // Remove internal fields and undefined values
+  const {_id, _rev, _createdAt, _updatedAt, _type, _version, ...rest} = JSON.parse(
+    JSON.stringify(document),
+  )
+
+  return rest
+}
+
+/**
+ * Compares two documents with the same schema type.
```
Comment (jordanl17) [inline packages/sanity/src/core/releases/tool/detail/review/DocumentDiff.tsx:21]: So simple yet so powerful! 🚀

File: packages/sanity/src/core/preview/useObserveDocument.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * @beta
+ *
+ * Observes a document by its ID and returns the document and loading state
+ * it will listen to the document changes.
+ */
+export function useObserveDocument<T extends SanityDocument>(
+  documentId: string,
+  schemaType: ObjectSchemaType,
+): {
+  baseDocument: T | null
```
[the remark this answers is not in the document]
Comment (pedrobonamin) [reply packages/sanity/src/core/preview/useObserveDocument.ts:17]: Yes, it should be `document`

File: packages/sanity/src/core/preview/useObserveDocument.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import {useDocumentPreviewStore} from '../store/_legacy/datastores'
+
+/**
+ * @internal
+ * @beta
+ *
+ * Observes a document by its ID and returns the document and loading state
+ * it will listen to the document changes.
+ */
+export function useObserveDocument<T extends SanityDocument>(
```
[the remark this answers is not in the document]
Comment (pedrobonamin) [reply packages/sanity/src/core/preview/useObserveDocument.ts:17]: Updated https://github.com/sanity-io/sanity/pull/7155/files#diff-ec0ff5e1eaf86dbdcaeee937037ff63df29303c0bdc6acd976ab4b997d2655e5

**Review by bjoerge — APPROVED** [review]
Great work!

**Pull request merged**